### PR TITLE
Improve analysis

### PR DESCRIPTION
**Pull request by Warchant**
- add shared_model folder to sonar analysis
- fixes in analysis steps (CMake)
- changes in codecov

### CODE REVIEW

File: cmake/analysis.cmake [hunk 1]
```diff
@@ -18,10 +18,12 @@ else()
   message(STATUS "Target cppcheck enabled")
   add_custom_target(cppcheck
     COMMAND ${CPPCHECK_BIN} --xml --xml-version=2 ${CMAKE_SOURCE_DIR}
-      -I ${CMAKE_SOURCE_DIR}/irohad/ -I ${CMAKE_SOURCE_DIR}/iroha-cli/
```
Comment (l4l) [inline cmake/analysis.cmake]: Why did you skip iroha-cli?

Comment (Warchant) [reply cmake/analysis.cmake]: cmake does not make "include_directories" on iroha-cli

Comment (Warchant) [reply cmake/analysis.cmake]: -I is "include folder"

Comment (l4l) [reply cmake/analysis.cmake]: What about this?:
https://github.com/hyperledger/iroha/blob/035ee547d4819829aea8b1937386216e1de0baa5/iroha-cli/CMakeLists.txt#L51-L53

Comment (Warchant) [reply cmake/analysis.cmake]: okay, did not notice this. thanks.

File: .codecov.yml [hunk 2]
```diff
@@ -1,28 +1,46 @@
...
+        if_ci_failed: error
+        only_pulls: false
+        flags: null
+        paths: null
     project:
-      default:
+      default: # measures whole project
+        # basic
+        target: auto
+        threshold: 0%
```
[the remark this answers is not in the document]
Comment (Warchant) [reply .codecov.yml]: "Allow coverage drop by 0% for success status" -- means that new code should not decrease  overall project coverage.

File: .codecov.yml [hunk 3]
```diff
@@ -1,28 +1,46 @@
...
+        base: auto
+      tests:
         target: auto
-        threshold: 0.05
+        paths: "tests/"  # only include coverage in "tests/" folder
```
[the remark this answers is not in the document]
Comment (Warchant) [reply .codecov.yml]: fxd

File: .codecov.yml [hunk 4]
```diff
@@ -1,28 +1,46 @@
...
-      tests:
+      libs:
         target: auto
-        paths: "test/"
-    changes:
-      default:
-        enabled: no
+        paths:
+          - "libs/"
+          - "schema/"
```
[the remark this answers is not in the document]
Comment (Warchant) [reply .codecov.yml]: I am also not sure, so I decided to put it into `libs` subgroup. I will remove schema  coverage, probably it is useless metric  for us.

File: cmake/analysis.cmake (hunk 1 above)
Comment (Warchant) [reply cmake/analysis.cmake]: Fxd

**Pull request merged**